### PR TITLE
Feature/include robot model

### PR DESCRIPTION
**Pull request by buschbapti**
This PR includes the `robot_model library` originally included in `robot_interface`. The Dockerfile installs the correct dependencies to build the library.

One things that concerns me is the time the library take to build. We should think on the Docker dependency to avoid rebuilding everything in dependent images.

### CODE REVIEW

**Comment by buschbapti** [comment]
I noticed I used the wrong version of the lib. It is now corrected

File: source/lib/robot_model/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+)
+
+install(TARGETS ${PROJECT_NAME}
+  	ARCHIVE DESTINATION lib
+  	LIBRARY DESTINATION lib
+  	RUNTIME DESTINATION bin
+)
+
+if (runtests)
+	  if (APPLE)
```
Comment (domire8) [inline source/lib/robot_model/CMakeLists.txt:62]: indentation?

File: source/lib/robot_model/include/robot_model/Exceptions/FrameNotFoundException.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#include <exception>
+
+namespace RobotModel
+{
+	namespace Exceptions
+	{
+		class FrameNotFoundException : public std::runtime_error
+		{
+		public:
+			explicit FrameNotFoundException(const std::string& frame_name) :
```
Comment (domire8) [inline source/lib/robot_model/include/robot_model/Exceptions/FrameNotFoundException.hpp:13]: indentation?

File: source/lib/robot_model/tests/testRobotModel.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#include <gtest/gtest.h>
+#include "robot_model/Model.hpp"
+
+using namespace RobotModel;
+
+TEST(LoadModel, PositiveNos)
+{
+	Model model("staubli", "/home/ros2/models/staubli_rx160/staubli_rx160.urdf");
```
Comment (domire8) [inline source/lib/robot_model/tests/testRobotModel.cpp:8]: I know testing is off but should this be removed or changed if we dont include the urdfs here?

**Comment by domire8** [comment]
I think the indentation should be corrected according to the style guide, like the rest of the modulo library.
Otherwise I'm happy if we can put it here, we'll find more improvements along the way.

About the docker build time. Maybe we could host an image that only contains modulo online somewhere and then build other dependent (ROS-)images on top. or is that not possible?

File: source/lib/robot_model/tests/testRobotModel.cpp (hunk 3 above)
Comment (buschbapti) [reply source/lib/robot_model/tests/testRobotModel.cpp:8]: yes that is a fair point. Either we remove it or include a test model somehow

**Comment by buschbapti** [comment]
Identation has been corrected and documentation cleaned up. There were some unused parameters that have been removed. Test has been removed for now.

**Review by eeberhard — APPROVED** [review]
I haven't tested this locally but it seems appropriate to me. Minor comments that have very little to do with the migration itself. 

For Docker I agree with Dominic, I think it's imperative that the base image in the Docker FROM command is always available pre-built somewhere. Whether that's docker hub or GitHub Container Repository we can discuss elsewhere, but it certainly would reduce build time and make it even easier to use.

File: source/lib/robot_model/include/robot_model/Model.hpp [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        Eigen::SparseMatrix<double> constraint_matrix_; ///< constraint matrix for the quadratic programming based inverse kinematics
+        Eigen::VectorXd lower_bound_constraints_; ///< lower bound matrix for the quadratic programming based inverse kinematics
+        Eigen::VectorXd upper_bound_constraints_; ///< upper bound matrix for the quadratic programming based inverse kinematics
+        std::shared_ptr<StateRepresentation::Parameter<double>> alpha_; ///< gain for the time optimization in the quadratic programming based inverse kinematics
+        std::shared_ptr<StateRepresentation::Parameter<double>> epsilon_; ///< minimal time for the time optimization in the quadratic programming based inverse kinematics
+        std::shared_ptr<StateRepresentation::Parameter<double>> linear_velocity_limit_; ///< maximum linear velocity allowed in the inverse kinematics
+        std::shared_ptr<StateRepresentation::Parameter<double>> angular_velocity_limit_; ///< maximum angular velocity allowed in the inverse kinematics
+        std::shared_ptr<StateRepresentation::Parameter<double>> proportional_gain_;
```
[the remark this answers is not in the document]
Comment (buschbapti) [reply source/lib/robot_model/include/robot_model/Model.hpp]: those are actually the unused parameters and have been removed

**Review by domire8 — APPROVED** [review]
good for me. we should think about the Docker image next

File: source/lib/robot_model/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      add_definitions(-DGTEST_USE_OWN_TR1_TUPLE)
+      add_definitions(-D__GLIBCXX__)
+    endif (APPLE)
+
+    add_subdirectory(lib/googletest)
+
+    enable_testing()
+
+  	# Include the gtest library. gtest_SOURCE_DIR is available due to
+  	# 'project(gtest)' above.
```
[the remark this answers is not in the document]
Comment (buschbapti) [reply source/lib/robot_model/CMakeLists.txt]: I think this is invoked in when doing

```
add_subdirectory(lib/googletest)
enable_testing()
```

at least it is working for the other lib. However, we should review a bit more the process for unit testing anyway. I am not happy with how it is done and included in the build process of the image at the moment.

File: source/lib/robot_model/include/robot_model/Model.hpp [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+        const std::string& get_urdf_path() const;
+
+        /**
+         * @brief Setter of the URDF path
+         * @param urdf_path the new value of the path
+         */
+        void set_urdf_path(const std::string& urdf_path);
+
+        /**
+         * @brief Gettor of the joint names attribute
```
[the remark this answers is not in the document]
Comment (buschbapti) [reply source/lib/robot_model/include/robot_model/Model.hpp]: corrected

**Pull request merged**